ts/src/test/scala/ch/epfl/ts/test/component/SimpleTraderTest.scala

```diff
@@ -0,0 +1,104 @@
+package ch.epfl.ts.test.component
+
+import scala.concurrent.duration.DurationInt
+import scala.language.postfixOps
+import org.scalatest.WordSpecLike
+import akka.actor.ActorSystem
+import akka.actor.Props
+import akka.testkit.TestActorRef
+import akka.testkit.TestKit
+import ch.epfl.ts.component.StartSignal
+import ch.epfl.ts.data.Currency
+import ch.epfl.ts.traders.SimpleFXTrader
+import com.typesafe.config.ConfigFactory
+import ch.epfl.ts.indicators.SMA
+import akka.testkit.EventFilter
+
+class SimpleTraderTest extends TestKit(ActorSystem("testSystem", ConfigFactory.parseString(
+  """
+  akka.loglevel = "DEBUG"
+  akka.loggers = ["akka.testkit.TestEventListener"]
+  """))) with WordSpecLike {
+
+  val traderId: Long = 123L
+  val symbol = (Currency.USD, Currency.CHF)
+  val volume = 1000.0
+  val shortPeriod = 5
+  val longPeriod = 30
+  val periods = List(5, 30)
+  val tolerance = 0.0002
+
+  val trader = TestActorRef(Props(classOf[SimpleFXTrader], traderId, symbol, shortPeriod, longPeriod, volume, tolerance))
+  trader ! StartSignal
+
+  "A trader " should {
+    "buy (20,3)" in {
+      within(1 second) {
+        EventFilter.debug(message = "buying 1000.0", occurrences = 1) intercept {
```

Comment:
In all expected `message`, use your `volume` variable rather than hardcoding this number each time.